wordpress-mobile/WordPress-Android
# Adds WPUtils as composite build

This PR is a follow up to #13001 and makes https://github.com/wordpress-mobile/WordPress-Utils-Android a binary dependency and adds the option to make it a composite build for local development. It also sets up the repository to remove the libs/utils subtree since it's no longer used.

Please make sure to check out #13001 for details on what composite build is and how it works. Since the merge of this PR will be a bit involved, I am not duplicating that information here to keep things brief. Also note that, the PR currently includes changes from #13001 since it's not yet merged, but once it is merged, the changes should stop showing up. (sometimes it doesn't in which case, please change the base branch to another branch and change it back to `develop` which hopefully resolves the issue)

**To test:**

* Copy `local-builds.gradle-example` as `local-builds.gradle` and update `localWPUtilsPath` to your liking
* Use `Navigate Class` shortcut to open `UrlUtils` which is in WPUtils and add a new static method such as a summation
* Use `Navigate Class` shortcut to open `ReaderPostLogic` (any other class would be fine) which is in WPAndroid and use that function
* Run the project and verify that it works 🎉 
* Comment out `localWPUtilsPath` in `local-builds.gradle` file and verify that the build fails for missing the new method from `UrlUtils`
* Discard the changes for `ReaderPostLogic`, build the app and verify that it works

### **How to merge this PR:**

1. Review and merge #13001.
2. Review & test this PR and approve it.
3. Prepare an announcement for the new workflow.
4. Remove the `libs/utils` folder and push that change. This should fix the CI failure. (this is not done yet to make it easier to review the changes)
5. Sync all the changes from WPAndroid's `origin/develop` with https://github.com/wordpress-mobile/WordPress-Utils-Android. This is very important since we are removing the `libs/utils` in this PR in the 3rd step.
6. Publish a new version of https://github.com/wordpress-mobile/WordPress-Utils-Android to Bintray (if necessary)
7. Update the `ext.wordPressUtilsVersion` in `build.gradle` to the latest version 
8. Re-test the PR.
9. Announce the good news 🥳 
10. Merge it! 🎉
